R/s3.R

```diff
@@ -177,7 +184,9 @@ s3_get <- function(ind_file, verbose = FALSE,
   if(verbose) {message("Downloading ", data_file, " from S3")}
   
   s3_config <- yaml::yaml.load_file(config_file)
-  aws.signature::use_credentials(profile = s3_config$profile)
+  if (use_local_aws_credentials) {
+    aws.signature::use_credentials(profile = s3_config$profile)
```

Comment:
What happens if you don't add these?  I think this is run already when `aws.signature` is loaded, and grabs the default profile from the default location if no environment vars are set, per the help file:

> use_credentials uses credentials from a profile stored in a credentials file to set the environment variables used by this package. It is called by default during package load if the AWS_ACCESS_KEY_ID and AWS_SECRET_ACCESS_KEY environment variables are not set.

But I don't think this would hurt anything to make sure the expected profile is being used, if you wanted to use something besides the default AWS profile and didn't already set the environment variable.

Reply:
If I don't add the `if` logic here, the s3 operations fail on ECS with something like "could not find /root/.aws/credentials".

I am a little bit hazy on exactly what is happening under the hood, but it seems that `aws.ec2metadata` may be providing `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY`, so that `aws.signature::read_credentials()` doesn't need to be called? Whereas if we call that function, it obscures the information from `aws.ec2metadata`?

In the ECS case, I did have to work around not having my region in a profile, which I did by [just setting it as an environment variable](https://github.com/jesse-ross/gage-conditions-gif/blob/awsify/aws/run-pipeline.sh#L3). Apart from region and credentials, do you know other important things which would go into the profile? I.e. do you think I need to be figure out a way to read the profile but not clobber the credentials when doing so?

Reply:
Right, yeah `use_credentials` won't work on EC2.  I guess what I'm asking is, is `use_credentials` still needed on a local machine, since it sounds like it runs when the aws.signature package is loaded already?  

I don't think there is other profile information you would need besides region.  I think its reasonable to expect a user to mostly handle credentials outside of this package if needed.

Reply:
Got it - yeah, it looks like if you're using a default profile, then there'd be no need for the `use_credentials()` call. But it's still nice to have this code for the case where the user wants to specify a non-default profile in the `s3_config.yml`.